### DISCUSSION TITLE
lxqt-archiver vs. Arqiver?

### DISCUSSION THREAD

**Discussion by probonopd** [tsujan/Arqiver#24] [Q&A] [answered]
Are 

* https://github.com/tsujan/Arqiver
* https://github.com/lxqt/lxqt-archiver

somehow related? Are both under active development, are there any philosophical differences between the two?
Which one would you recommend us to ship in [helloSystem](https://hellosystem.github.io/docs/)?

**Comment by tsujan** [comment, marked as answer]
lxqt-archiver belongs to LXQt and is a Qt fork of engrampa — its core code is based on GLib and it uses `libfm-qt` — while Arqiver is an DE-independent, Qt-based front-end for `bsdtar` (as it's called in Linux), `7z` and `gzip`. They're both under active development: lxqt-archiver by the LXQt team (me included) and Arqiver by me.

Their main difference is that lxqt-archiver uses GLib to handle the command-line options of its backends, while Arqiver does it by using Qt. Because of that, Arqiver has a much more compact code.

**Reply by probonopd** [reply]
Thanks. From this description it is clear to me that I like Arqiver better.

**Reply by tsujan** [reply]
Arqiver should work under *BSD as it is — it even has an extra option under *BSD — but if you encounter any problem, please tell me!

**Reply by probonopd** [reply]
Has been working well for me:

![image](https://user-images.githubusercontent.com/2480569/111875245-14638480-8999-11eb-975a-d00af0cd9367.png)

Now need to get it into Filer...
https://github.com/helloSystem/Filer/issues/54

**Reply by tsujan** [reply]
> Now need to get it into Filer...

In LXQt, we use this (inside `/usr/share/libfm-qt/archivers.list`):

```
[arqiver]
create=arqiver --sa %U
extract=arqiver --sx %U
extract_to=arqiver --ax %U
mime_types=application/x-xz-compressed-tar;application/x-lzma-compressed-tar;application/x-bzip-compressed-tar;application/x-bzip;application/x-tar;application/x-compressed-tar;application/zip;application/gzip;image/svg+xml-compressed;application/x-gzpdf;application/bzip;application/x-7z-compressed;application/vnd.debian.binary-package;application/x-archive;application/x-cpio;application/x-rpm;application/x-source-rpm;application/x-cd-image;application/x-raw-disk-image;application/x-xar;application/x-java-archive;
supports_uris=true
```

**Comment by tsujan** [comment]
> Which one would you recommend us to ship in helloSystem?

If you've forked libfm-qt as an independent library in `helloSystem`, you could also fork lxqt-archiver. Otherwise, I think forking of Arqiver might be easier.

However, please note that lxqt-archiver supports more archive formats. For example, I included `application/x-zstd-compressed-tar` in both Arqiver and lxqt-archiver but added `application/zstd` only to lxqt-archiver (didn't find it worth the extra code that was needed by Arqiver). I think there are other examples too.
